### PR TITLE
Fixing okhttp hang when TLS nego fails.

### PR DESCRIPTION
**Pull request by nmittler**
Negotiation failure results in a RuntimeException, which is not properly  handled by the okhttp code, resulting in the client hanging.

Refactored the code to shutdown the transport when TLS negotiation fails.

### CODE REVIEW

**Comment by nmittler** [comment]
@madongfly @ejona86 PTAL

**Comment by ejona86** [comment]
@nmittler LGTM

**Comment by madongfly** [comment]
@nmittler LGTM

**Comment by nmittler** [comment]
@madongfly PTAL .. I've cleaned up the exception handling a bit.

File: okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java [hunk 1]
```diff
@@ -587,11 +594,10 @@ public void run() {
         // Read until the underlying socket closes.
         while (frameReader.nextFrame(this)) {
         }
-      } catch (IOException ioe) {
-        // We send GoAway here because OkHttp wraps many protocol errors as IOException.
+      } catch (Throwable t) {
```
Comment (madongfly) [inline okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java]: Just Exception? we may not want to catch Error?

Comment (nmittler) [inline okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java]: done.

**Comment by nmittler** [comment]
@madongfly PTAL

**Comment by madongfly** [comment]
@nmittler LGTM

**Comment by nmittler** [comment]
Cherry-picked as 0cd28dd

**Pull request closed without merging**